### DISCUSSION TITLE
5.1 树与二叉树基础

### DISCUSSION THREAD

**Discussion by giscus** [itcharge/AlgoNote#172] [Comments]
# 5.1 树与二叉树基础

树与二叉树的基础知识 1. 树简介 1.1 树的定义 树（Tree）：由 n≥0 个节点与节点之间的关系组成的有限集合。当 n=0 时称为空树，当 n>0 时称为非空树。 之所以把这种数据结构称为「树」是因为这种数据结构看起来就像是一棵倒挂的树，也就是说数据结构中的「树」是根朝上，而叶朝下的。如下图所示。 树树 「树」具有以下的特点： 有且仅有一个节点...

[https://algo.itcharge.cn/05_tree/05_01_tree_basic/](https://algo.itcharge.cn/05_tree/05_01_tree_basic/)

**Comment by MortyCats** [comment]
写的真好！我觉得比javaguide清晰多了，既不过分简单，又有合适的是实现案例和练习链接。 此外，这里对树的分类我觉得做的很好，javaguide那边还专门把堆heap分了一类，其实就是一种完全二叉树

**Reply by MortyCats** [reply]
看到这样好的作品，真是赏心悦目、心旷神怡，感谢作者的努力，辛苦！

**Comment by zrjjysf** [comment]
完全二叉树的定义语句是错的，约束不足，得加上下面特点的第三句才能构成完整的定义。

**Comment by Mike-7777777** [comment]
"节点分类" 示意图中箭头应该是从A出发,而不是从C出发. 指向文字, 而非指向空气.

**Reply by itcharge** [reply]
感谢指正，内容稍后修正。
